back/boxtribute_server/business_logic/statistics/crud.py

```diff
@@ -3,17 +3,56 @@
 from peewee import JOIN, SQL, fn
 
 from ...db import db
-from ...enums import HumanGender, TaggableObjectType
+from ...enums import BoxState, HumanGender, TaggableObjectType
 from ...models.definitions.beneficiary import Beneficiary
 from ...models.definitions.box import Box
+from ...models.definitions.history import DbChangeHistory
 from ...models.definitions.location import Location
 from ...models.definitions.product import Product
 from ...models.definitions.product_category import ProductCategory
+from ...models.definitions.size import Size
 from ...models.definitions.tag import Tag
 from ...models.definitions.tags_relation import TagsRelation
+from ...models.definitions.transaction import Transaction
 from ...models.utils import convert_ids
 
 
+def _generate_dimensions(*names, facts):
+    """Return a dictionary holding information (ID, name) about dimensions with
+    specified names.
+    """
+    dimensions = {}
+
+    if "product" in names:
+        product_ids = {f["product_id"] for f in facts}
+        dimensions["product"] = (
+            Product.select(Product.id, Product.name)
+            .where(Product.id << product_ids)
+            .dicts()
+        )
+
+    if "category" in names:
+        dimensions["category"] = ProductCategory.select(
+            ProductCategory.id, ProductCategory.name
+        ).dicts()
+
+    if "tag" in names:
```

Comment:
I think it would be nice if you could also return the color here in general.

Reply:
addressed in #998